src/resources/packages/wizard/components/tabs/settings/tab.tsx

```diff
@@ -0,0 +1,229 @@
+	{ label: __('Saturday', 'the-events-calendar'), value: '6' },
+];
+
+const SettingsContent = ({moveToNextTab, skipToNextTab}) => {
+	const visitedFields = useSelect(select => select(SETTINGS_STORE_KEY).getVisitedFields() || {} );
+	const setVisitedField = useDispatch(SETTINGS_STORE_KEY).setVisitedField;
+	const { currency, timezone_string, date_format, start_of_week, timezones, currencies }: { currency: string, timezone_string: string, date_format: string, start_of_week: number, timezones: Record<string, Record<string, string>>, currencies: Record<string, { symbol: string, name: string }> } = useSelect(
+		(select) => {
+			const store = select(SETTINGS_STORE_KEY);
+			return {
+				currency: store.getSetting('currency'),
+				timezone_string: store.getSetting('timezone_string'),
+				date_format: store.getSetting('date_format'),
+				start_of_week: store.getSetting('start_of_week'),
+				timezones: store.getSetting('timezones'),
+				currencies: store.getSetting('currencies'),
+			};
+		},
+		[]
+	);
+	const [ currencyCode, setCurrency ] = useState( currency );
+	const [ timeZone, setTimeZone ] = useState( timezone_string );
+	const [ dateFormat, setDateFormat ] = useState( date_format || dateFormatOptions[0].value );
+	const [ weekStart, setWeekStart ] = useState( start_of_week || 0 );
+	const [canContinue, setCanContinue] = useState(false);
+
+	let timeZoneMessage = __("Please ensure your time zone is correct.", 'the-events-calendar');
+
+	if ( ! timezone_string ) {
+		timeZoneMessage = __("Please select your time zone.", 'the-events-calendar');
+	} else if ( timezone_string.includes('UTC') ) {
+		timeZoneMessage = __('Please select your time zone as UTC offsets are not supported.', 'the-events-calendar');
+	}
+
+	// Create tabSettings object to pass to NextButton.
+	const tabSettings = {
+		currency: currencyCode,
+		timezone_string: timeZone,
+		date_format: dateFormat,
+		start_of_week: weekStart,
+		currentTab: 2, // Include the current tab index.
+	};
+
+	useEffect(() => {
+		// Define the event listener function.
+		const handleChange = (event) => {
+			setVisitedField(event.target.id);
+		};
+
+		const fields = document.getElementById('settingsPanel')?.querySelectorAll('input, select, textarea');
+		fields?.forEach((field) => {
+			field.addEventListener('change', handleChange);
+		});
+
+		return () => {
+			fields?.forEach((field) => {
+				field.removeEventListener('change', handleChange);
+			});
+		};
+	}, []);
+
+	// Compute whether the "Continue" button should be enabled
+	useEffect(() => {
+		// Since most of these are selects, we just ensure there is a value.
+		const fieldsToCheck = {
+			"currencyCode": currencyCode,
+			"timeZone": isValidTimeZone(),
+			"dateFormat": dateFormat,
+			"weekStart": weekStart,
+			'visit-at-least-one': hasVisitedHere(),
+		};
+
+		setCanContinue(Object.values(fieldsToCheck).every((field) => !!field));
+	}, [currencyCode, timeZone, dateFormat, weekStart, visitedFields]);
```

Comment:
We shouldn't need it as well. Define invidual handlers for the changes of the fields